messages/src/protocols/out_of_band/invitation.rs

```diff
@@ -23,7 +23,11 @@ pub struct OutOfBandInvitation {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub handshake_protocols: Option<Vec<MessageType>>, // TODO: Make a separate type
     pub services: Vec<ServiceOob>,
-    #[serde(rename = "requests~attach")]
+    #[serde(
+        default = "Attachments::new",
+        skip_serializing_if = "Attachments::is_empty",
+        rename = "requests~attach"
+    )]
```

Comment:
I think you can say just `#[serde(defaul)]`, as that calls the `Default::default()` of the type, so `Attachments::default()`.